### ISSUE TITLE
Can not build meilisearch from source on amazon-linux-2023

### ISSUE THREAD

**Issue by wangshijun** [meilisearch/meilisearch#4868]
**Describe the bug**

Can not build meilisearch from source on amazon-linux-2023

**To Reproduce**

I just followed https://www.meilisearch.com/docs/learn/self_hosted/install_meilisearch_locally#local-installation to build meilisearch on amazon linux 2023, but it failed

**Expected behavior**

The build should succeed without any error

**Screenshots**

```
$ cargo build --release
   Compiling time v0.3.34
   Compiling httpdate v1.0.2
   Compiling digest v0.10.7
   Compiling lindera-dictionary-builder v0.31.0
   Compiling regex v1.10.4
   Compiling num-complex v0.4.5
   Compiling rand_distr v0.4.3
   Compiling webpki-roots v0.25.3
   Compiling atomic v0.5.3
   Compiling uuid v1.7.0
   Compiling dyn-stack v0.10.0
   Compiling raw-cpuid v10.7.0
   Compiling pulp v0.18.9
   Compiling half v2.4.0
   Compiling unicase v2.6.0
error[E0282]: type annotations needed for `Box<_>`
  --> /home/ec2-user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/time-0.3.34/src/format_description/parse/mod.rs:83:9
   |
83 |     let items = format_items
   |         ^^^^^
...
86 |     Ok(items.into())
   |              ---- type must be known at this point
   |
help: consider giving `items` an explicit type, where the placeholders `_` are specified
   |
83 |     let items: Box<_> = format_items
   |              ++++++++

   Compiling cpufeatures v0.2.12
   Compiling seq-macro v0.3.5
   Compiling local-waker v0.1.3
   Compiling bitflags v2.5.0
   Compiling serde_urlencoded v0.7.1
   Compiling lindera-cc-cedict-builder v0.31.0
   Compiling lindera-unidic-builder v0.31.0
   Compiling lindera-ipadic-neologd-builder v0.31.0
   Compiling lindera-ko-dic-builder v0.31.0
   Compiling lindera-ipadic-builder v0.31.0
   Compiling fxhash v0.2.1
   Compiling synstructure v0.13.1
   Compiling ucd-trie v0.1.6
For more information about this error, try `rustc --explain E0282`.
error: could not compile `time` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

**Meilisearch version:**

v1.9.0

**Additional context**

```
rustc --version
rustc 1.80.1 (3f5fd8dd4 2024-08-06)
```

```
uname -a
Linux staging2024 6.1.79-99.167.amzn2023.x86_64 #1 SMP PREEMPT_DYNAMIC Tue Mar 12 18:15:55 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```
```

**Comment by wangshijun** [comment]
The build succeeds with v1.10.0-rc.2

**Comment by curquiza** [comment]
Hello!

This is probably fixed due to this PR in v1.10
https://github.com/meilisearch/meilisearch/pull/4706/files

I close this issue as done since it will be fixed in v1.10
